.github/workflows/ci.yml

```diff
@@ -59,22 +59,25 @@ jobs:
           cache: 'maven'
       - name: Configure Problem Matchers
         run: echo "::add-matcher::.github/problem-matcher.json"
+      - name: Download all Trino dependencies
+        run: |
+          $RETRY $MAVEN dependency:go-offline
       - name: Maven Checks
         run: |
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
-          $RETRY $MAVEN clean install -B --strict-checksums -V -T C1 -DskipTests -P ci -pl '!:trino-server-rpm'
+          $MAVEN clean install -B --strict-checksums -V -T C1 -DskipTests -P ci -pl '!:trino-server-rpm'
```

Comment:
why remove retry? Failure can be because of other reasons too than just Maven central issues (OOM for example).

Reply:
Same for `verify`.

Reply:
(It's a question - no change requested yet - I'm probably missing something)

Reply:
I'm assuming builds are deterministic, so the only reason to leave retries is when dependencies have to be downloaded due to a Maven cache miss. I've removed retries from the `maven-checks` job since it downloads all dependencies first, but I plan to leave retries in other jobs.

Reply:
Makes sense. We can revisit (and fix issues) if we see compilation fail in future.